hpcc-nagios-tools/hpcc_centralized_nagios.py

```diff
@@ -0,0 +1,707 @@
+#!/usr/bin/env python3
+
+import lxml.etree as ET
+import subprocess
+import paramiko
+import getpass
+import sys
+import os
+
+Environment = input('Enter Environment Name > ')
+Address = input('Enter IP > ')
+ProdOrNot = input(
+    'Please specify if this environment is production by typing prod or if it is a non-production by typing non-prod > ')
+xxx = getpass.getpass()
+user = input('Enter username > ')
+xml_local = '/tmp/environment.xml'
+xml_remote = '/etc/HPCCSystems/environment.xml'
+
+if ProdOrNot == "prod":
+    HostDefinition = "generic-prod-hpcc"
+    ServiceDefinition = "hpcc-prod-service"
+elif ProdOrNot == "non-prod":
+    HostDefinition = "generic-nonprod-hpcc"
+    ServiceDefinition = "hpcc-nonprod-service"
+else:
+    print("Invalid Option")
+    sys.exit(1)
+
+
+class XML():
+    def __init__(self):
+        self.xml = self
+
+    def getXML(self):
+        host = paramiko.Transport((Address, 22))
+        host.connect(username=user, password=xxx)
+        sftp = paramiko.SFTPClient.from_transport(host)
+        sftp.get(xml_remote, xml_local)
+
+
+class XMLParser():
+
+    def __init__(self):
+        self.parsing = self
+
+    def IPs(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        ip_addresses = []
+        for i in doc_root.findall('.//DafilesrvProcess/Instance'):
+            ip = i.attrib['netAddress']
+            ip_addresses.append(ip)
+        return ip_addresses
+
+    def NumberOfServers(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        ip_addresses = []
+        for i in doc_root.findall('.//DafilesrvProcess/Instance'):
+            ip = i.attrib['netAddress']
+            ip_addresses.append(ip)
+        return ip_addresses
+
+    def dali(self):
+        ParseXML = ET.parse(xml_local)
+        doc_root = ParseXML.getroot()
+        for i in doc_root.findall('.//DaliServerProcess/Instance'):
+            DaliIP = i.attrib['netAddress']
+        return DaliIP
```

Comment:
All following probably can be kept just one function with a parameter of (component process name). If dali, etc we only want to return the last one just use the last index to reference it.

Reply:
I implemented the suggested changes, I just need to run a couple of tests.